staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1572,3 +1567,24 @@ func buildOpenAPIModelsForApply(staticOpenAPISpec *spec.Swagger, crd *apiextensi
 	}
 	return models, nil
 }
+
+func findCRD(crdName string, crds []*apiextensionsv1.CustomResourceDefinition) (*apiextensionsv1.CustomResourceDefinition, string, error) {
+	var crdClusterName string
+	var crd *apiextensionsv1.CustomResourceDefinition
+
+	for _, aCRD := range crds {
+		if aCRD.Name != crdName {
+			continue
+		}
+		if crd == nil {
+			crd = aCRD
```

Comment:
can there be more than one crd with the given name? Or could we just return here?

Reply:
I went back and looked at the feature-logical-clusters-1.22 branch and what I'm seeing there is if you're trying to do a wildcard list or watch across logical clusters, it fails if 2 logical clusters have the same CRD (by name) but with different specs.

Reply:
ok, then maybe we need this logic. `findCRDWithUniqueSchema` or something like that would fit.

Reply:
Surely we need some negotiation / merging...
But this initial PR was mainly a fix for the basic case.
Letting the case when your have several different schemas for a later PR. This seems clearly related to the future work on API import / export, and the work @ncdc is doing related to discovery and CRDs